File: tsp-solver-frontend/src/app/model/request/spp/generator/GraphResult.ts
import {NodeRequest} from "../NodeRequest";
import {RoadTypeRequest} from "../RoadTypeRequest";
import {EdgeRequest} from "../EdgeRequest";

export class GraphResult {
  nodes: NodeRequest[];
  roadTypes: RoadTypeRequest[];
  edges: EdgeRequest[];

  constructor(nodes: NodeRequest[], roadTypes: RoadTypeRequest[], edges: EdgeRequest[]) {
    this.nodes = nodes;
    this.roadTypes = roadTypes;
    this.edges = edges;
  }

}

File: tsp-solver-backend/src/main/java/pl/edu/pbs/spp/generator/GraphResult.java
package pl.edu.pbs.spp.generator;

import lombok.AllArgsConstructor;
import lombok.Data;
import pl.edu.pbs.spp.Edge;
import pl.edu.pbs.spp.Node;
import pl.edu.pbs.spp.RoadType;

import java.util.List;

@Data
@AllArgsConstructor
public class GraphResult {

    private List<Node> nodes;
    private List<RoadType> roadTypes;
    private List<Edge> edges;

}
